CRM/Contact/BAO/Contact/Location.php

```diff
@@ -40,16 +40,22 @@ class CRM_Contact_BAO_Contact_Location {
    *
    * @param bool $isPrimary
    * @param int $locationTypeID
+   * @param bool $isBilling
    *
    * @return array
    *   Array of display_name, email, location type and location id if found, or (null,null,null, null)
    */
-  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL) {
+  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL, $isBilling = FALSE) {
     $primaryClause = NULL;
     if ($isPrimary) {
       $primaryClause = " AND civicrm_email.is_primary = 1";
     }
 
+    // CRM-20362 retrieving Billing email address.
+    if ($isBilling) {
+      $primaryClause = " AND civicrm_email.is_billing = 1";
```

Comment:
> So at the code level - I think the fall back to primary should be happening in the getEmailDetails function not in the calling code and I would be reluctant to merge this without tests.

I've thought of it, but let's say if someone wants to check if the billing address is set of a contact or not, getEmailDetails will always return values (true). (Falling back to primary email address)

> I would probably also wrap that function in a function getBillingEmailDetails rather than just add another input variable.

Agreed, It's good to have different function rather than updating current one. I'll do and push the changes if all agrees.